File: examples/ja_stats.py
# This file contains statistics functions written with Japanese comments
# 数値リストの基本統計を計算するシンプルなスクリプト

# サンプル数値のリストを定義する
numbers = [4, 8, 15, 16, 23, 42]

# 全ての数値の合計を計算する
total = sum(numbers)

average = total / len(numbers)  # 合計を個数で割って平均値を計算する

minimum = min(numbers)  # リスト内の最小値を見つける
maximum = max(numbers)  # リスト内の最大値を見つける

# 結果をコンソールに出力する
print(f"Numbers: {numbers}")
print(f"Sum:     {total}")
print(f"Average: {average:.2f}")
print(f"Min:     {minimum}")
print(f"Max:     {maximum}")

File: examples/ja_stats.rs
// This file contains statistics functions written with Japanese comments

/*
 * 基本的な統計を計算する関数群。
 * i32スライスを入力として受け取る。
 */

/* ------ 計算関数 ------ */

// 整数スライスの合計を計算する
fn sum(numbers: &[i32]) -> i32 {
    numbers.iter().sum() // イテレータを使って合計を求める
}

// スライスの最小値を返す
fn minimum(numbers: &[i32]) -> i32 {
    *numbers.iter().min().unwrap() // 最小値を取得して参照を外す
}

// スライスの最大値を返す
fn maximum(numbers: &[i32]) -> i32 {
    *numbers.iter().max().unwrap() // 最大値を取得して参照を外す
}

fn main() {
    // サンプル数値の配列を定義する
    let numbers = [4, 8, 15, 16, 23, 42];

    let total = sum(&numbers); /* 合計を計算する */
    let average = total as f64 / numbers.len() as f64; // 平均値を計算する

    /* ------ 結果を出力する ------ */
    println!("数値:   {:?}", numbers);
    println!("合計:   {}", total);
    println!("平均:   {:.2}", average);
    println!("最小値: {}", minimum(&numbers));
    println!("最大値: {}", maximum(&numbers));
}
